packages/zent/src/grid/Grid.tsx

```diff
@@ -734,14 +722,12 @@ export class Grid<Data = any> extends PureComponent<
 
   onSelectChange = (selectedRowKeys: string[], data: Data | Data[]) => {
     const { datasets, selection } = this.props;
-    const onSelect: IGridSelection<Data>['onSelect'] = get(
-      selection,
-      'onSelect'
-    );
+    const onSelect: IGridSelection<Data>['onSelect'] = selection?.onSelect;
 
-    if (isFunction(onSelect)) {
-      const selectedRows = (datasets || []).filter((row, i) =>
-        includes(selectedRowKeys, this.getDataKey(row, i))
+    if (typeof onSelect === 'function') {
+      const selectedRows = (datasets || []).filter(
+        (row, i) =>
+          (selectedRowKeys ?? []).indexOf(this.getDataKey(row, i)) !== -1
```

Comment:
datasets?.filter((row, i) => selectedRowKeys?.indexOf(this.getDataKey(row, i)) !== -1) ?? []